src/test/java/integ/basic/BasicProcess.java

```diff
@@ -0,0 +1,107 @@
+package integ.basic;
+
+import static integ.basic.BasicProcess.INPUT;
+import static integ.basic.BasicProcess.STATE_ID_NEXT_1;
+import static integ.basic.BasicProcess.STATE_ID_NEXT_2;
+
+import io.xdb.core.process.Process;
+import io.xdb.core.state.AsyncState;
+import io.xdb.core.state.AsyncStateOptions;
+import io.xdb.core.state.StateDecision;
+import io.xdb.core.state.StateMovement;
+import io.xdb.core.state.StateSchema;
+import io.xdb.gen.models.CommandRequest;
+import io.xdb.gen.models.CommandWaitingType;
+import lombok.NonNull;
+import org.junit.jupiter.api.Assertions;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BasicProcess implements Process {
+
+    public static final String STATE_ID_NEXT_1 = "STATE_ID_NEXT_1";
+    public static final String STATE_ID_NEXT_2 = "STATE_ID_NEXT_2";
+    public static final Integer INPUT = 11;
+
+    @Override
+    public @NonNull StateSchema getStateSchema() {
```

Comment:
I guess @NonNull is not required right? It will look a little cleaner if it's not required

Reply:
@nonnull will make sure the StateSchema will not be bull. So that we don't need to check `if (getStateSchema() == null)` in case the user `return null` for the method. Adding it can reduce the checking logic on the SDK side.

Reply:
I feel like we don't have to enforce this. Our SDK can be smart enough to consider a null as empty schema.

Reply:
Got it. Consider a null as empty schema sounds like a more user-friendly approach. I will remove the nonnull annotation and deal with the null cases.